use_case_guide/real_time_dashboards.rst

```diff
@@ -282,74 +237,87 @@ which often appear.
+approximately; it takes a surprisingly small amount of space to tell you
+approximately how many unique elements are in a set. Its accuracy can be
+adjusted. We'll use ones which, using only 1280 bytes, will be able to
+count up to tens of billions of unique visitors with at most 2.2% error.
 
 An equivalent problem appears if you want to run a global query, such as the number of
-unique ip addresses which visited any of your client's sites over the last month. Without
-HLLs this query involves shipping lists of ip addresses from the workers to the coordinator for
+unique IP addresses which visited any of your client's sites over the last month. Without
+HLLs this query involves shipping lists of IP addresses from the workers to the coordinator for
 it to deduplicate. That's both a lot of network traffic and a lot of computation. By using
 HLLs you can greatly improve query speed.
 
-First you must install the hll extension; `the github repo
+First you must install the HLL extension; `the github repo
 <https://github.com/aggregateknowledge/postgresql-hll>`_ has instructions. Next, you have
 to enable it:
 
 .. code-block:: sql
 
-  -- this part must be run on all nodes
-  CREATE EXTENSION hll;
-
-  -- this part runs on the coordinator
-  ALTER TABLE http_request_1min ADD COLUMN distinct_ip_addresses hll;
+  -- Run on all nodes ------------------------------------
 
-To make hll functions work with distributed tables, we should define hll aggregate functions
-with the name `sum` on all nodes and let Citus use these functions for distributed queries:
+  CREATE EXTENSION hll;
 
-.. code-block:: sql
+  -- Run on the coordinator node -------------------------
 
-  -- alias for hll_add_agg
+  -- allow SUM to work on hashvals (alias for hll_add_agg)
   CREATE AGGREGATE sum(hll_hashval) (
-       SFUNC = hll_add_trans0,
-       STYPE = internal,
-       FINALFUNC = hll_pack
+    SFUNC = hll_add_trans0,
+    STYPE = internal,
+    FINALFUNC = hll_pack
   );
 
-  -- alias for hll_union_agg
-  CREATE AGGREGATE sum(hll)(
-      sfunc = hll_union_trans,
-      stype = internal,
-      finalfunc = hll_pack
-  );
+.. note::
 
-When doing our rollups, we can now aggregate sessions into an hll column with queries
-like this:
+  This is not necessary on Citus Cloud, which has HLL already installed,
+  along with other useful :ref:`cloud_extensions`.
+
+Now we're ready to track IP addresses in our rollup with HLL. First
+add a column to the rollup table.
 
 .. code-block:: sql
 
-  SELECT
-    site_id, date_trunc('minute', ingest_time) as minute,
-    sum(hll_hash_text(ip_address)) AS distinct_ip_addresses
-  FROM http_request
-  WHERE date_trunc('minute', ingest_time) = date_trunc('minute', now())
-  GROUP BY site_id, minute;
+  ALTER TABLE http_request_1min ADD COLUMN distinct_ip_addresses hll;
+
+Next use our custom aggregation to populate the column. Just add it
+to the query in our rollup function:
+
+.. code-block:: diff
+
+  @@ -1,10 +1,12 @@
```

Comment:
I thought the diff format makes it easier to understand how small the necessary change is. Our old explanation was like, "See this query? Imagine melding it into the rollup."

The drawback of the diff is that it doesn't provide anything to directly copy and run.